Make Desktop version unit testable

ArtjomP opened the pull request:
Default AppName and CompanyName have been changed from empty to values provoking to change them. Because it's impossible to set assembly's CompanyName attribute for UnitTesting process.

ArtjomP committed:
Default AppName and CompanyName have been changed from empty to values. Because it's impossible to set assembly's CompanyName attribute for UnitTesting process.

jaderfeijo commented:
@picolyl please merge this. I am currently being prevented from Unit Testing some code I wrote using this library because of the same issue described above.

picolyl commented:
I think it is better to have an error if there is no name specified.
I provided a means to specify directly instead.
How about using it when testing?

((FolderPathImplementation)FolderPath.Current).AppName = "TestApp";
((FolderPathImplementation)FolderPath.Current).CompanyName = "TestCompany";

... Test Code Here ...

ArtjomP commented:
Great! I like it! It’s a usual way for all Xamarin Plugins. I just need some time to check it on my battle field

ArtjomP committed:
Default values for AppName and CompanyName have been removed. Possibility to use your own IFolderPath implementation has been added (ex UnitTests) .

ArtjomP commented:
I've updated the code to support using user's IFolderPath implementation as it does in Xamarin Plugins (http://motzcod.es/post/159267241302/unit-testing-plugins-for-xamarin). Also I've described it in the readme.md. I've tested both ways in my project. Both work well.

ArtjomP commented:
I've updated the code. I removed default values for AppName & CompanyName. Also I added possibility to set FoldertPath.Current for using custom IFolderPath implementation (mock for example).
Both ways (((FolderPathImplementation)FolderPath.Current) & FolderPath.Current set) were tested and documented in readme.md.

jaderfeijo commented:
This is great @ArtjomP, thanks for implementing this. It is just what I needed.

ArtjomP committed:
Description has been updated

jaderfeijo commented:
@picolyl thank you so much for merging this. Now that it's merged, when do you think this will become available as part of an update to the library. Is there a release roadmap?

picolyl commented:
Nuget Package Updated.

jaderfeijo commented:
Awesome! Thank you so much 😊
